app/src/components/MealCard/MealCard.jsx

```diff
@@ -0,0 +1,230 @@
+      .then((response) => {
+        const formattedData = response.data.totalReserved;
+        setRightFormatData(formattedData);
+      })
+      .catch((error) => {
+        console.error("Error fetching reservation data:", error);
+      });
+  };
+
+  useEffect(() => {
+    fetchReservationData();
+  }, [id]);
+
+  const [rightFormatData, setRightFormatData] = useState(0);
+  const [showPopup, setShowPopup] = useState(false);
+  const [showRatePop, setShowRatePop] = useState(false);
+  const [contactName, setContactName] = useState("");
+  const [numberOfGuests, setNumberOfGuests] = useState(0);
+  const [contactPhoneNumber, setContactPhoneNumber] = useState(0);
+  const [contactEmail, setContactEmail] = useState("");
+
+  // Rating state variables
+  const [rating, setRating] = useState(0);
+  const [ratingComment, setRatingComment] = useState("");
+  const [ratingTitle, setRatingTitle] = useState("");
+
+  const handleNameInput = (e) => {
+    setContactName(e.target.value);
+  };
+  const handleGuestNumberInput = (e) => {
+    setNumberOfGuests(Number(e.target.value));
+  };
+  const handlePhoneInput = (e) => {
+    setContactPhoneNumber(e.target.value);
+  };
+  const handleEmailInput = (e) => {
+    setContactEmail(e.target.value);
+  };
+
+  const handleFormSubmit = (e) => {
+    e.preventDefault();
+    setShowPopup(false);
+    axios
+      .post("http://localhost:3001/api/reservations/add", formData)
+      .then((response) => {
+        console.log("Reservation added successfully:", response.data);
+        alert("Reservation added successfully!");
+        fetchReservationData();
+      })
+      .catch((error) => {
+        console.error("Error adding meal:", error);
+        alert("Error adding meal: " + error.message);
+      });
+  };
+
+  const handleRatingCommentInput = (e) => {
+    setRatingComment(e.target.value);
+  };
+
+  const handleRatingTitleInput = (e) => {
+    setRatingTitle(e.target.value);
+  };
+
+  const handleRatingSubmit = (e) => {
+    e.preventDefault();
+    setShowRatePop(false);
+
+    axios
+      .post("http://localhost:3001/api/reviews/add", ratingData)
+      .then((response) => {
+        console.log("Rating added successfully:", response.data);
+        alert("Rating added successfully!");
+      })
+      .catch((error) => {
+        console.error("Error adding rating:", error);
+        alert("Error adding rating: " + error.message);
+      });
+  };
+
+  const formData = {
+    contact_name: contactName,
+    number_of_guests: numberOfGuests,
+    contact_phonenumber: contactPhoneNumber,
+    contact_email: contactEmail,
+    meal_id: id,
+  };
+
+  const ratingData = {
+    meal_id: id,
+    stars: rating,
+    description: ratingComment,
+    title: ratingTitle,
+  };
+
+  return (
+    <>
+      <div className={styles.cardContainer}>
+        <div className={`${styles.card} ${styles.uClearfix}`}>
+          <div className="test">
```

Comment:
className test? :)

Reply:
hahaha thats from brain-burn times when I was trying things to make it work :D , gonna fix it

Reply:
I know the feeling :joy: